src/unstract/sdk/file_storage/fs_impl.py

```diff
@@ -308,25 +323,80 @@ def get_hash_from_file(self, path: str) -> str:
             h = sha256()
             b = bytearray(128 * 1024)
             mv = memoryview(b)
-            with self._open_with_no_buffering(path=path) as f:
+            with self.fs.open(path) as f:
                 while n := f.readinto(mv):
                     h.update(mv[:n])
             return str(h.hexdigest())
+        except FileNotFoundError as e:
+            raise e
         except Exception as e:
             raise FileOperationError(str(e)) from e
 
-    def json_dump(self, path, mode, encoding, data, **kwargs):
+    def json_dump(
+        self,
+        path: str,
+        data: dict[str, Any],
+        **kwargs: dict[Any, Any],
+    ):
         """Dumps data into the given file specified by path.
 
         Args:
             path (str): Path to file where JSON is to be dumped
-            mode (str): write modes
-            encoding (str): Encoding to be used while writing the file
-            data (bytes|str): data to be written to the file
+            data (dict): Object to be written to the file
             **kwargs (dict): Any other additional arguments
         """
         try:
-            with self.fs.open(path=path, mode=mode, encoding=encoding) as f:
+            with self.fs.open(path=path, mode="w", encoding="utf-8") as f:
```

Comment:
@gaya3-zipstack It would give more granular control if we add encoding in the method signature.

Reply:
@gaya3-zipstack just to be extensible in the future, I'd suggest adding an encoding but with a default of utf-8. I noticed a request related to different encoding in the LLMW py client previously

Reply:
@gaya3-zipstack An optional param would be help here, instead of passing it with hard coded encoding.

Reply:
Will change it in the next PR.